src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,85 @@
+package duke;
+
+import duke.command.Command;
+import duke.command.Parser;
+import duke.command.Storage;
+import duke.command.TaskList;
+import duke.command.Ui;
+
+import java.util.Scanner;
+
+/**
+ * A robot that can respond to user's input, and help user store tasks
+ */
+public class Duke {
+    private Command command;
+
+    /**
+     * Constructor of Duke class.
+     *
+     * @param filePath The filePath of the file storing the data.
+     */
+    public Duke(String filePath) {
+        TaskList taskList = new TaskList();
+        Ui ui = new Ui(taskList);
+        Storage storage = new Storage(filePath, taskList);
+        this.command = new Command(taskList, ui, storage);
+    }
+
+    /**
+     * Start running the program.
+     */
+    public void run() {
+        //Read events stored in the input List
+        command.loadSavedTasks();
+
+        command.showLogo();
+
+        Scanner inputScanner = new Scanner(System.in);
+        String task = inputScanner.nextLine();
+        while (!Parser.isBye(task)) {
+            try {
+                switch (Parser.judgeType(task)) {
+                case TODO:
+                    command.addNewTodo(task);
+                    break;
+                case EVENT:
+                    command.addNewEvent(task);
+                    break;
+                case DEADLINE:
+                    command.addNewDeadline(task);
+                    break;
+                case DONE:
+                    command.setTaskDone(task);
+                    break;
+                case LIST:
+                    command.showList();
+                    break;
+                case DELETE:
+                    command.deleteTask(task);
+                    break;
+                case FIND:
+                    command.findTask(task);
+                    break;
+                case UNKNOWN:
+                    throw new Exception("Cannot Understand");
```

Comment:
Perhaps it's better to have customized classes for different kinds of exceptions, thereby customizing the exception message for each class. Then in the lines below you won't need to check if the exception message is equal to some message and then print out something else; instead you just need to throw an instance of that specific class of customized exception and the corresponding message can also be displayed.